### PR TITLE
mm: Add mm_uninitialize to release the resource

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Impact
New API

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

File: fs/procfs/fs_procfsmeminfo.c [hunk 1]
```diff
@@ -582,4 +582,28 @@ void procfs_register_meminfo(FAR struct procfs_meminfo_entry_s *entry)
...
+ * Input Parameters:
+ *   entry - Describes the entry to be unregistered.
+ *
+ ****************************************************************************/
+
+void procfs_unregister_meminfo(FAR struct procfs_meminfo_entry_s *entry)
+{
+  FAR struct procfs_meminfo_entry_s **cur;
+
+  for (cur = &g_procfs_meminfo; *cur != NULL; cur = &cur->next)
```
Comment (pkarashchenko) [inline fs/procfs/fs_procfsmeminfo.c:600]:
```suggestion
  for (cur = g_procfs_meminfo; cur != NULL; cur = cur->next)
```

Comment (pkarashchenko) [inline fs/procfs/fs_procfsmeminfo.c:600]:
```suggestion
  for (cur = &g_procfs_meminfo; *cur != NULL; cur = &(*cur)->next)
```

Comment (xiaoxiang781216) [reply fs/procfs/fs_procfsmeminfo.c]: Done.

**Comment by xiaoxiang781216** [comment]
@pkarashchenko could you merge this PR?

**Review by masayuki2009 — APPROVED** [review]

**Pull request merged**